.circleci/config.yml

```diff
@@ -0,0 +1,139 @@
+defaults: &defaults
+  working_directory: ~/loga
+  steps:
+    - checkout
+    - &restore_gems
+      restore_cache:
+        keys:
+          - v2-gems-{{ checksum "Gemfile" }}-{{ checksum "Appraisals"}}-{{ .Environment.CIRCLE_JOB }}
+          - v2-gems-{{ checksum "Gemfile" }}
+    - &bundle_install
+      run: bundle install --jobs 4 --path=vendor/bundle --retry 3
+    - run: bundle exec appraisal install
+    - &cache_gems
+      save_cache:
+        key: v2-gems-{{ checksum "Gemfile" }}-{{ checksum "Appraisals"}}-{{ .Environment.CIRCLE_JOB }}
+        paths:
+          - ~/loga/gemfiles/vendor/bundle
+          - ~/loga/vendor/bundle
+    - attach_workspace:
+        at: ./tmp
+    - run: RACK_ENV=development bundle exec appraisal rspec
+    - run: RACK_ENV=production bundle exec appraisal rspec
+    - run: ./tmp/cc-test-reporter format-coverage -t simplecov -o $(ruby -e 'puts "tmp/codeclimate.#{RUBY_VERSION}.json"') coverage/.resultset.json
+    - persist_to_workspace:
+        root: tmp
+        paths:
+          - codeclimate.*.json
+    - store_artifacts:
+        path: coverage
+
+version: 2
+jobs:
+  build:
+    working_directory: ~/loga
+    docker:
+      - image: circleci/ruby:2.5.1
+    steps:
+      - run:
+          name: Download cc-test-reporter
+          command: |
+            mkdir -p tmp/
+            curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > ./tmp/cc-test-reporter
+            chmod +x ./tmp/cc-test-reporter
+      - persist_to_workspace:
+          root: ./tmp
+          paths:
+            - cc-test-reporter
+  upload-coverage:
+    working_directory: ~/loga
```

Comment:
🙋 Suggest setting this default earlier in the file also. You can `<<:` it into `&defaults` above.